res/controllers/Hercules-DJControl-Inpulse-300-script.js

```diff
@@ -44,76 +60,239 @@ DJCi300.scratchShiftMultiplier = 4;
 DJCi300.bendScale = 1.0;
 
 // Other scratch related options
-DJCi300.kScratchActionNone = 0;
+DJCi300.kScratchActionBend = 0;
 DJCi300.kScratchActionScratch = 1;
 DJCi300.kScratchActionSeek = 2;
-DJCi300.kScratchActionBend = 3;
+
+// Pad modes
+DJCi300.padModeNone = -1;
+DJCi300.padModeHotcue = 0;
+DJCi300.padModeRoll = 1;
+DJCi300.padModeSlicer = 2;
+DJCi300.padModeSampler = 3;
+DJCi300.padModeToneplay = 4;
+DJCi300.padModeFX = 5;
+DJCi300.padModeSlicerloop = 6;
+DJCi300.padModeBeatjump = 7;
+
+// Slicer connections
+let slicerSizeConnection1;
+let slicerTrackConnection1;
+let slicerBeatConnection1;
+let slicerTrackConnection2;
+let slicerSizeConnection2;
+let slicerBeatConnection2;
 
 DJCi300.vuMeterUpdateMaster = function(value, _group, _control) {
-    value = (value * 122) + 5;
+    value = (value * 125);
     midi.sendShortMsg(0xB0, 0x40, value);
     midi.sendShortMsg(0xB0, 0x41, value);
 };
 
 DJCi300.vuMeterUpdateDeck = function(value, group, _control, _status) {
-    value = (value * 122) + 5;
-    var status = (group === "[Channel1]") ? 0xB1 : 0xB2;
+    value = (value * 125);
+    const status = (group === "[Channel1]") ? 0xB1 : 0xB2;
     midi.sendShortMsg(status, 0x40, value);
 };
 
 DJCi300.init = function() {
     // Scratch button state
-    DJCi300.scratchButtonState = true;
+    DJCi300.scratchButtonState = {
+        1: true,
+        2: true
+    };
```

Comment:
why this over an array or better names?
```suggestion
    DJCi300.scratchButtonState = {
        "[Channel1]": true,
        "[Channel2]": true
    };
```
```suggestion
    DJCi300.scratchButtonState = [true, true];
```

Reply:
I originally did it this way so that you can use 1 and 2 directly (based on the decks) instead of 0 and 1. But yeah, I think it would be easier to just use `[Channel1]` and `[Channel2]`.

Reply:
both of the patterns I suggested are established, so choose whatever you prefer.